spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ThrowingExceptions.java

```diff
@@ -42,10 +94,121 @@ public void sawOpcode(int seen) {
                     reportBug("THROWS_METHOD_THROWS_RUNTIMEEXCEPTION", getXMethod());
                 }
             }
+        } else if (exceptionThrown != null &&
+                seen == Const.INVOKEVIRTUAL ||
+                seen == Const.INVOKEINTERFACE ||
+                seen == Const.INVOKESTATIC) {
+            XMethod calledMethod = getXMethodOperand();
+            if (calledMethod == null) {
+                return;
+            }
+
+            String[] thrownExceptions = calledMethod.getThrownExceptions();
+            if (thrownExceptions != null && Arrays.stream(thrownExceptions)
+                    .anyMatch(s -> s.replace('/', '.').equals(exceptionThrown))) {
+                exceptionThrown = null;
+            }
+
         }
     }
 
     private void reportBug(String bugName, XMethod method) {
-        bugReporter.reportBug(new BugInstance(this, bugName, NORMAL_PRIORITY).addClass(this).addMethod(method));
+        bugReporter.reportBug(new BugInstance(this, bugName, LOW_PRIORITY).addClass(this).addMethod(method));
+    }
+
+    private boolean parentThrows(Method method, String exception) {
+        return parentThrows(klass, method, exception);
+    }
+
+    private boolean parentThrows(JavaClass clazz, Method method, String exception) {
+        JavaClass ancestor;
+        boolean throwsEx = false;
+        try {
+            ancestor = clazz.getSuperClass();
+            if (ancestor != null) {
+                Optional<Method> superMethod = Arrays.stream(ancestor.getMethods())
+                        .filter(m -> method.getName().equals(m.getName()) && sameSignature(method, m))
+                        .findAny();
+                if (superMethod.isPresent()) {
+                    throwsEx = Arrays.stream(superMethod.get().getExceptionTable().getExceptionNames())
+                            .anyMatch(s -> exception.equals(s));
+                } else {
+                    throwsEx = parentThrows(ancestor, method, exception);
+                }
+            }
+
+            for (JavaClass intf : clazz.getInterfaces()) {
+                Optional<Method> superMethod = Arrays.stream(intf.getMethods())
+                        .filter(m -> method.getName().equals(m.getName()) && sameSignature(method, m))
+                        .findAny();
+                if (superMethod.isPresent()) {
+                    throwsEx |= Arrays.stream(superMethod.get().getExceptionTable().getExceptionNames())
+                            .anyMatch(s -> exception.equals(s));
+                } else {
+                    throwsEx |= parentThrows(intf, method, exception);
+                }
+            }
+        } catch (ClassNotFoundException e) {
+            AnalysisContext.reportMissingClass(e);
+        }
+        return throwsEx;
+    }
+
+    private boolean sameSignature(Method child, Method parent) {
```

Comment:
This method is quite complex, consider making it package-private and write several unit test cases to cover each expected case.

Reply:
Or, I have not tried it yet but it could be simplify this implementation with help from `org.objectweb.asm.Type` or `edu.umd.cs.findbugs.ba.SignatureParser`.